crates/core/tedge_mapper/src/c8y/converter.rs

```diff
@@ -317,16 +307,12 @@ where
     fn register_external_device(
         &mut self,
         tedge_event: &ThinEdgeEvent,
-        messages: &mut Vec<Message>,
+        mqtt_messages: &mut Vec<Message>,
     ) -> bool {
-        if let Some(c_id) = tedge_event.source.clone() {
+        if let Some(child_id) = tedge_event.source.clone() {
             // Create the external source if it does not exists
-            if !self.children.contains(&c_id) {
-                self.children.insert(c_id.clone());
-                messages.push(Message::new(
-                    &Topic::new_unchecked(SMARTREST_PUBLISH_TOPIC),
-                    format!("101,{c_id},{c_id},thin-edge.io-child"),
-                ));
+            if !self.children.contains_key(&child_id) {
+                is_external_device_registered(child_id, &mut self.children, mqtt_messages);
```

Comment:
The check `!self.children.contains_key` is already done by `is_external_device_registered`

Reply:
Agreed, the extra check is required here because this function has to return `true` or `false` based on the device has to be created or not.